### ISSUE TITLE
image in distance calculation: got 0.0

### ISSUE THREAD

**Issue by hainm** [Amber-MD/cpptraj#163]
This is pytraj issue, I just bring for discussion.

This is the situation: I strip water, ion, ... from my explicit water simulation (nobox) and strip  prmtop too. However, I forgot to set no box for Topology, so it still have PBC info.

When loading to pytraj, perform distance calculation, I got array of [0.0 ,..], which is different from cpptraj's output. After trial and error, I found that I need to specify image=False (noimage in cpptraj) and the output from pytraj matches to cpptraj's ones.

I know why this happen: because I pass the Box info from Topology rather from Coordinfo in traj to ActionList setup.

I just want to discuss as user' perspective: I made a fake (in memory) trajectory (with no PBC) and  accidentally use a Topology with Box info. How can I avoid to get false output?

What do you think about letting cpptraj doing the checking (Frame has box or not vs Topopology has box or not) or letting pytraj doing the checking?

**Comment by drroe** [comment]
> What do you think about letting cpptraj doing the checking (Frame has box or not vs Topopology has box or not) or letting pytraj doing the checking?

I have a simple solution - don't use box information from the Topology, period. In my opinion things like unit cell information should be associated with coordinates (this was the driving idea behind development of the `CoordinateInfo` class). The *only* times that box info from the Topology is needed is for Amber coordinates since typically only the box lengths are stored in the trajectory, so the Topology is used to figure out the angles (orthorhombic vs truncated octahedron), and Amber topology, where the box information is written as `beta A B C` (and this is only necessary because of the first case).

In summary, use `CoordinateInfo` to determine what coordinates look like.

**Comment by hainm** [comment]
should add live demo

```python
In [7]: traj = pt.iterload('strip.nc', 'strip_with_box.prmtop', frame_slice=(0, 3))

In [8]: traj.top
Out[8]: <Topology: 527 atoms, 16 residues, 1 mols, PBC with box type = truncoct>

In [9]: for f in traj:
    print(f.box)
   ...:     
<Box: nobox, (x, y, z, alpha, beta, gamma) = (0.0, 0.0, 0.0, 0.0, 0.0, 0.0)>
<Box: nobox, (x, y, z, alpha, beta, gamma) = (0.0, 0.0, 0.0, 0.0, 0.0, 0.0)>
<Box: nobox, (x, y, z, alpha, beta, gamma) = (0.0, 0.0, 0.0, 0.0, 0.0, 0.0)>

In [10]: pt.distance(traj, ':3 :10')
Out[10]: array([ 0.,  0.,  0.])

In [11]: pt.distance(traj, ':3 :10', image=False)
Out[11]: array([ 18.69267083,  17.93175541,  19.04500565])
```

**Comment by hainm** [comment]
> In summary, use CoordinateInfo to determine what coordinates look like.

sounds good for me.
